src/speed.c

```diff
@@ -0,0 +1,134 @@
+#include "speed.h"
+
+/**
+ * Maximum acceleration and deceleration.
+ *
+ * - Linear acceleration is defined in meters per second squared.
+ * - Linear deceleration is defined in meters per second squared.
+ * - Angular acceleration is defined in radians per second squared.
+ */
+static volatile float linear_acceleration;
+static volatile float linear_deceleration;
+static volatile float angular_acceleration;
+static volatile float max_linear_speed;
+static volatile float max_angular_speed;
+static volatile float turn_linear_speed;
+static volatile float turn_radius;
+static volatile float t0;
+static volatile float t1;
+
+const float linear_acceleration_defaults[NUM_MODES] = {5., 0., 0.};
+const float linear_deceleration_defaults[NUM_MODES] = {5., 0., 0.};
+const float angular_acceleration_defaults[NUM_MODES] = {32. * PI, 0., 0.};
+const float max_linear_speed_defaults[NUM_MODES] = {.5, 0., 0.};
+const float max_angular_speed_defaults[NUM_MODES] = {3 * PI, 0., 0.};
+const float turn_linear_speed_defaults[NUM_MODES] = {0.448, 0., 0.};
+const float turn_radius_defaults[NUM_MODES] = {0.11, 0., 0.};
+const float t0_defaults[NUM_MODES] = {166, 0., 0.};
+const float t1_defaults[NUM_MODES] = {260, 0., 0.};
+
+float get_linear_acceleration(void)
+{
+	return linear_acceleration;
+}
+
+void set_linear_acceleration(float value)
+{
+	linear_acceleration = value;
+}
+
+float get_linear_deceleration(void)
+{
+	return linear_deceleration;
+}
+
+void set_linear_deceleration(float value)
+{
+	linear_deceleration = value;
+}
+
+float get_angular_acceleration(void)
+{
+	return angular_acceleration;
+}
+
+void set_angular_acceleration(float value)
+{
+	angular_acceleration = value;
+}
+
+float get_max_linear_speed(void)
+{
+	return max_linear_speed;
+}
+
+void set_max_linear_speed(float value)
+{
+	max_linear_speed = value;
+}
+
+float get_max_angular_speed(void)
+{
+	return max_angular_speed;
+}
+
+void set_max_angular_speed(float value)
+{
+	max_angular_speed = value;
+}
+
+float get_turn_linear_speed(void)
+{
+	return turn_linear_speed;
+}
+
+void set_turn_linear_speed(float value)
+{
+	turn_linear_speed = value;
+}
+
+float get_turn_radius(void)
+{
+	return turn_radius;
+}
+
+void set_turn_radius(float value)
+{
+	turn_radius = value;
+}
+
+float get_t0(void)
```

Comment:
I'd say `get_turn_t0`, for consistency.

Reply:
(same for `set_t0` and `*_t1` functions)

Reply:
ok